tests/unit/test_redfish.py

```diff
@@ -555,11 +565,69 @@ def mock_get_response(uri):
         mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storage/STOR1")
         mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storage/STOR2")
 
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish_client")
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.redfish_utilities.collections.get_collection_ids"  # noqa: E501
+    )
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.redfish_utilities.systems.get_system_ids"
+    )
+    def test_non_standard_storage_uri_name(
+        self, mock_get_system_ids, mock_get_collection_ids, mock_redfish_client
+    ):
+        """Test non-standard name for "Storage" in URI for storage controller and drives.
+
+        Eg: /redfish/v1/Systems/S1/Storages
+        """
+        mock_redfish_obj = Mock()
+        mock_get_system_ids.return_value = ["s1"]
+        mock_get_collection_ids.return_value = ["STOR1"]
+        mock_redfish_client.return_value = mock_redfish_obj
+
+        def mock_get_response(uri):
+            response = Mock()
+            if "Systems/s1/Storages/STOR1/Drives/d11" in uri:
+                response.dict = {
+                    "Id": "d11",
+                    "Status": {"Health": "OK", "State": "Enabled"},
+                }
+            elif "Systems/s1/Storages/STOR1" in uri:
+                response.dict = {
+                    "StorageControllers": [
+                        {
+                            "MemberId": "sc0",
+                            "Status": {"Health": "OK", "State": "Enabled"},
+                        }
+                    ],
+                    "Drives": [
+                        {"@odata.id": "/redfish/v1/Systems/s1/Storages/STOR1/Drives/d11"},
+                    ],
+                }
+            # response for GET request to /redfish/v1/Systems/<sys_id>/
+            elif "Systems" in uri:
+                response.dict = {"Storage": {"@odata.id": "/redfish/v1/Systems/sX/Storages"}}
+            return response
+
+        mock_redfish_obj.get.side_effect = mock_get_response
+
+        # storage controller
+        with RedfishHelper(Mock()) as helper:
+            _sc_count, _sc_data = helper.get_storage_controller_data()
```

Comment:
Why we didn't check the expected value of sc_count and sc_data? 

The testing logic between test_non_standard_storage_uri_name and test_get_storage_controller_data_success should be the same. We only need to simulate different `mock_get_response`.

Reply:
> Why we didn't check the expected value of sc_count and sc_data?
> The testing logic between test_non_standard_storage_uri_name and test_get_storage_controller_data_success should be the same. We only need to simulate different mock_get_response.

I was doing this to not duplicate the code being tested with `test_get_storage_controller_data_success`. The purpose of `test_non_standard_storage_uri_name` was only to check in case there's a non standard storage name being used, we were actually querying the right endpoint for the redfish API. The actual storage controller data or count doesn't matter because if we're sure about the endpoint, then we're testing the ability to obtain the data, count from `test_get_storage_controller_data_success`.

Reply:
Yes, I agree with @Pjack, testing the end product of that function is necessary. However, if you do want to test if you can get the correct storage name, then you might want to make the logic [here](https://github.com/canonical/prometheus-hardware-exporter/blob/6b1ce0880714a2a6e672145d11e2cc7bcd554bdb/prometheus_hardware_exporter/collectors/redfish.py#L273) as a separate function, and test it.

Reply:
But that being said, I think you don't need to remove that assertion, but add a check about the end result is fine.

Reply:
The main purpose of this testing is to make sure our call to redfish can compatible with different storage name.

Just want to point out url checking is the major checking item here.

The return object is not what we care about here, although you can say it indirect prove it's working and adding one more checking here is not a big deal